### PR TITLE
[JBDS-4227] cleaning up code

### PR DESCRIPTION
**Pull request by adietish**
# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

File: plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/controllers/CDKLaunchController.java [hunk 1]
```diff
@@ -153,99 +154,123 @@ private boolean isInitialized(ILaunchConfigurationWorkingCopy wc) throws CoreExc
...
 
-		IProcess process = addProcessToLaunch(p, launch,s);
-		linkTerminal(p);
-		
-		IDebugEventSetListener debug = getDebugListener(new IProcess[]{process}, launch);
-		DebugPlugin.getDefault().addDebugEventListener(debug);
-		if( beh != null ) {
-			beh.putSharedData(AbstractStartJavaServerLaunchDelegate.PROCESS, process);
-			beh.putSharedData(AbstractStartJavaServerLaunchDelegate.DEBUG_LISTENER, debug);
+	private void verifyUser(final ControllableServerBehavior beh, String userKey) throws CoreException {
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/controllers/CDKLaunchController.java]: weird, Sonar is clearly wrong here. #verifyUser is being called at line 205

File: plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/controllers/CDKLaunchController.java [hunk 2]
```diff
@@ -153,99 +154,123 @@ private boolean isInitialized(ILaunchConfigurationWorkingCopy wc) throws CoreExc
...
 		} catch(IOException ioe) {
 			CDKCoreActivator.pluginLog().logError(ioe);
 			beh.setServerStopped();
 			throw new CoreException(new Status(IStatus.ERROR, CDKCoreActivator.PLUGIN_ID, ioe.getMessage(), ioe));
 		}
-		
-		if( p == null ) {
+	}
+
+	private void verifyPassword(final ControllableServerBehavior beh, String passKey) throws CoreException {
```
Comment (adietish) [inline plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/adapter/controllers/CDKLaunchController.java]: weird, Sonar is clearly wrong here. #verifyPassword is being called at line 206

**Comment by bdshadow** [comment]
@adietish they seem to have lots of problems with unused private methods in sonarQube: https://jira.sonarsource.com/browse/MMF-462 However, all of them should be fixed already. I'll try to investigate this, and will create a bug for them in case i won't find anything

**Comment by adietish** [comment]
@bdshadow great, thx!
maybe you can have a look at this cleanup PR and accept/refuse it?

**Comment by bdshadow** [comment]
@adietish ok!

**Comment by adietish** [comment]
@bdshadow ok, let's hope that they chime in and follow up on this bug in sonar. Would reduce the "noise".
Could have review this PR? Would be great if we could get it in before the freeze this evening.

**Comment by bdshadow** [comment]
@adietish looks good to me. Do we need to test it better during the next sprint? Although PR looks good, isn't it dangerous to merge so many changes before the code freeze?

**Comment by adietish** [comment]
@bdshadow there should be no functional changes, so I'd go ahead and merge it, given the tests still pass.
@jeffmaury ?

**Comment by bdshadow** [comment]
@adietish ok, then let's merge it

**Comment by bdshadow** [comment]
testPR

**Comment by bdshadow** [comment]
@adietish @jeffmaury seems i can't test merged PR with _testPR_ comment.
However i changed the Jenkins job, according SonarQube team advice. These strange reports should be fixed (just don't know how to test it again). So in case any similar problems appear again, please, ping me!

**Comment by adietish** [comment]
@nickboldt do you know why @bdshadow cannot trigger jenkins via testPR?

**Comment by adietish** [comment]
@bdshadow what did you change?

**Comment by nickboldt** [comment]
bdshadow is listed in the whitelist for the github PR builder config in the job, https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/Devstudio_pull_requests/job/jbosstools-openshift-Pull-Request so he should be able to fire a "testPR" command at the process.

Is he also a member of the jbosstools github organization? If not, he should probably be added.

**Comment by adietish** [comment]
@nickboldt @bdshadow is member of the jbosstools org in github as far as I can see. @bdshadow looks like everything is in place so that you should be able to fire jenkins up. what makes you think that you can't?

**Comment by nickboldt** [comment]
To test the results of the MERGED PR, you would go here:

https://jenkins.mw.lab.eng.bos.redhat.com/hudson/view/DevStudio/view/DevStudio_Master/job/jbosstools-openshift_master/changes ... where this commit was built.

Then go here for the resulting update site:

http://download.jboss.org/jbosstools/neon/snapshots/builds/jbosstools-openshift_master/latest/all/repo/

**Comment by adietish** [comment]
@nickboldt thx!

**Comment by bdshadow** [comment]
@nickboldt thank you! So it turned out as i said above that already merged PR can't be "testPR"ed as usual.
@adietish I've added `compile` maven goal between `clean` and `sonar:sonar` goals. So the error was because the analysis was being done for NOT recompiled classes

**Pull request merged**